src/m365/pp/commands/tenant/tenant-settings-set.ts

```diff
@@ -0,0 +1,257 @@
+        autocomplete: ['true', 'false']
+      },
+      {
+        option: '--disableNewsletterSendout [disableNewsletterSendout]',
+        autocomplete: ['true', 'false']
+      },
+      {
+        option: '--disableEnvironmentCreationByNonAdminusers [disableEnvironmentCreationByNonAdminusers]',
+        autocomplete: ['true', 'false']
+      },
+      {
+        option: '--disablePortalsCreationByNonAdminusers [disablePortalsCreationByNonAdminusers]',
+        autocomplete: ['true', 'false']
+      },
+      {
+        option: '--disableSurveyFeedback [disableSurveyFeedback]',
+        autocomplete: ['true', 'false']
+      },
+      {
+        option: '--disableTrialEnvironmentCreationByNonAdminusers [disableTrialEnvironmentCreationByNonAdminusers]',
+        autocomplete: ['true', 'false']
+      },
+      {
+        option: '--disableCapacityAllocationByEnvironmentAdmins [disableCapacityAllocationByEnvironmentAdmins]',
+        autocomplete: ['true', 'false']
+      },
+      {
+        option: '--disableSupportTicketsVisibleByAllUsers [disableSupportTicketsVisibleByAllUsers]',
+        autocomplete: ['true', 'false']
+      },
+      {
+        option: '--disableDocsSearch [disableDocsSearch]',
+        autocomplete: ['true', 'false']
+      },
+      {
+        option: '--disableCommunitySearch [disableCommunitySearch]',
+        autocomplete: ['true', 'false']
+      },
+      {
+        option: '--disableBingVideoSearch [disableBingVideoSearch]',
+        autocomplete: ['true', 'false']
+      },
+      {
+        option: '--shareWithColleaguesUserLimit [shareWithColleaguesUserLimit]'
+      },
+      {
+        option: '--disableShareWithEveryone [disableShareWithEveryone]',
+        autocomplete: ['true', 'false']
+      },
+      {
+        option: '--enableGuestsToMake [enableGuestsToMake]',
+        autocomplete: ['true', 'false']
+      },
+      {
+        option: '--disableAdminDigest [disableAdminDigest]',
+        autocomplete: ['true', 'false']
+      },
+      {
+        option: '--disableDeveloperEnvironmentCreationByNonAdminUsers [disableDeveloperEnvironmentCreationByNonAdminUsers]',
+        autocomplete: ['true', 'false']
+      },
+      {
+        option: '--disableBillingPolicyCreationByNonAdminUsers [disableBillingPolicyCreationByNonAdminUsers]',
+        autocomplete: ['true', 'false']
+      },
+      {
+        option: '--disableChampionsInvitationReachout [disableChampionsInvitationReachout]',
+        autocomplete: ['true', 'false']
+      },
+      {
+        option: '--disableSkillsMatchInvitationReachout [disableSkillsMatchInvitationReachout]',
+        autocomplete: ['true', 'false']
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+
+        const regexNumber: RegExp = new RegExp(/\d/g);
+        if (args.options.shareWithColleaguesUserLimit && !regexNumber.test(args.options.shareWithColleaguesUserLimit)) {
```

Comment:
Let's use the built in function `isNaN(...)`. I wonder if the value should also be greater or equal to 1.

Reply:
After checking, it seems like the API uses an UInt. So let's just check that the value is greater or equal to 0.